lib/SILGen/SILGenProlog.cpp

```diff
@@ -432,6 +433,9 @@ void SILGenFunction::emitProlog(AnyFunctionRef TheClosure,
         SILLocation unreachableLoc(param);
         unreachableLoc.markAsPrologue();
         B.createUnreachable(unreachableLoc);
+        
+        auto &ctx = SGM.M.getASTContext();
+        ctx.Diags.diagnose(TheClosure.getBody()->getStartLoc(), diag::unreachable_code_uninhabited_param_note);
```

Comment:
Is it possible to reference the location of the specific parameter that's unreachable, and include its name in the diagnostic here? It'd be nice to say "'param' is of type 'Never' so this function body can never be executed" or something like that.

Reply:
Done :)